Sources/NIO/SystemDarwin.swift

```diff
@@ -0,0 +1,454 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+//
+//  This file contains code that ensures errno is captured correctly when doing
+//  syscalls and no ARC traffic can happen inbetween that *could* change the
+//  errno value before we were able to read it.  It's important that all static
+//  methods are declared with `@inline(never)` so it's not possible any ARC
+//  traffic happens while we need to read errno.
+//
+//  Created by Norman Maurer on 11/10/17.
+//
+
+#if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
+
+@_exported
+import Darwin.C
+
+import CNIODarwin
+
+internal typealias MMsgHdr = CNIODarwin_mmsghdr
+
+private func isBlacklistedErrno(_ code: Int32) -> Bool {
+    switch code {
+    case EFAULT, EBADF:
+        return true
+    default:
+        return false
+    }
+}
+
+private func preconditionIsNotBlacklistedErrno(err: CInt, where function: String) -> Void {
+    // strerror is documented to return "Unknown error: ..." for illegal value so it won't ever fail
+    precondition(!isBlacklistedErrno(err), "blacklisted errno \(err) \(String(cString: strerror(err)!)) in \(function))")
+}
+
+internal extension IOResult where T: FixedWidthInteger {
+  var result: T {
+    switch self {
+    case .processed(let value):
+      return value
+    case .wouldBlock(_):
+      fatalError("cannot unwrap IOError")
+    }
+  }
+}
```

Comment:
The indentation is off here. Also, why is this `internal` when the Linux version is `fileprivate`?

Reply:
Because of a very special reason: not building on Linux ;-).  The extension is used elsewhere, and so needs to be `internal`.

Reply:
😁 Hah ok then.